### PR TITLE
Add Logger performance test

### PR DESCRIPTION
**Pull request by ayush-sb**
# Pull Request

WIP: Add initial performance test for logger

### CODE REVIEW

File: tests/performance_tests/test_perf_logger.py [hunk 1]
```diff
@@ -22,15 +23,49 @@
...
+        "Initiating positron containment field",
+        "Initializing multi-phase quantum singularity containment field",
+        "Deploying ionized plasma thrusters for interstellar travel",
+        "Calibrating neural interface for optimal performance",
+        "Engaging gravimetric warp drive for faster-than-light travel",
+        "Activating tachyon pulse generator for temporal manipulation" "Activating shields",
+        "Charging plasma cannon",
+        "Deploying tractor beam",
+        "Initializing warp drive",
+        "Engaging hyperdrive",
```
Comment (cjdsellers) [inline tests/performance_tests/test_perf_logger.py:62]: :clap:

File: tests/performance_tests/test_perf_logger.py [hunk 2]
```diff
@@ -22,15 +23,49 @@
...
+    ]
 
-        for i in range(20):
-            logger_adapter.error(f"{i}")
+    def run():
+        for i in range(100_000):
+            message = random.choice(messages)
+            # unique log messages to prevent caching during string conversion
```
Comment (cjdsellers) [inline tests/performance_tests/test_perf_logger.py:68]: Good catch

**Comment by ayush-sb** [comment]
Here's the test results on the new logger
image

Same test with old logger, for comparison. The new logger is about 15% faster than the old one on my machine
image

**Comment by ayush-sb** [comment]
@cjdsellers You can merge it, thank you

**Pull request merged**